cost/azure/rightsize_managed_disks/azure_rightsize_managed_disks.pt

```diff
@@ -0,0 +1,1135 @@
+      newResourceType = fittingDiskName
+      newResourceTypeRedundancy = currentDiskRedundancy
+      newResourceTypeGiB = fittingDiskData.sizeGiB
+      newResourceTypeIOPS = fittingDiskData.IOPS
+      newResourceTypeThroughput = fittingDiskData.throughput
+      newResourceTypeMonthlyPrice = fittingDiskPrice.pricePerUnit
+      newResourceTypePriceCurrency = fittingDiskPrice.currencyCode
+    } else continue;
+
+    tags = []
+    if (typeof(disk.tags) == 'object') {
+      _.each(Object.keys(disk.tags), function(key) {
+        tags.push([key, "=", disk.tags[key]].join(''))
+      })
+    }
+
+    result.recommendations.push({
+      accountID: disk.subscriptionID,
+      accountName: disk.subscriptionName,
+      resourceGroup: disk.resourceGroup,
+      resourceName: disk.resourceName,
+      resourceID: disk.id,
+      tags: tags,
+      resourceType: currentDiskName,
+      resourceTypeRedundancy: currentDiskRedundancy,
+      resourceTypeGiB: currentDiskData.sizeGiB,
+      resourceTypeIOPS: currentDiskData.IOPS,
+      resourceTypeThroughput: currentDiskData.throughput,
+      resourceTypeMonthlyPrice: currentDiskPrice.pricePerUnit,
+      resourceTypePriceCurrency: currentDiskPrice.currencyCode,
+      newResourceType: newResourceType,
+      newResourceTypeRedundancy: newResourceTypeRedundancy,
+      newResourceTypeGiB: newResourceTypeGiB,
+      newResourceTypeIOPS: newResourceTypeIOPS,
+      newResourceTypeThroughput: newResourceTypeThroughput,
+      newResourceTypeMonthlyPrice: newResourceTypeMonthlyPrice,
+      newResourceTypePriceCurrency: newResourceTypePriceCurrency,
+      resourceKind: disk.resourceType,
+      region: disk.region,
+      monthlySavings: monthlySavings.toFixed(2),
+      savingsCurrency: currentDiskPrice.currencyCode,
+      iopsPctMaximum: Math.round(disk.iopsConsumedPctMax),
+      iopsPctAverage: Math.round(disk.iopsConsumedPctAvg),
+      iopsThreshold: param_min_used_disk_iops_pct,
+      throughputPctMaximum: Math.round(disk.bandwithConsumedPctMax),
+      throughputPctAverage: Math.round(disk.bandwithConsumedPctAvg),
+      throughputThreshold: param_min_used_disk_throughput_pct,
+      sizeGB: disk.diskSizeGB,
+      usedGBPct: disk.diskUsedGB != undefined ? Math.round(disk.diskUsedGB * 100 / disk.diskSizeGB) : "Unknown",
+      diskSpaceThreshold: ds_min_used_disk_space_pct,
+      lookbackPeriod: param_stats_lookback,
+      service: disk.service.split("/")[0],
+    })
+
+    result.monthlySavings += monthlySavings
+  }
+
+  if (result.monthlySavings >= param_min_savings) {
+    result.monthlySavings = result.monthlySavings.toFixed(2)
+    result.raiseIncident = true
+  }
+EOS
+end
+
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "pol_azure_rightsize_managed_disks" do
+  validate $ds_disk_rightsizing_recommendations do
```

Comment:
We should do a validate_each instead and have the policy check on each instance in a list. See other recommendations policies, such as the Azure Unused Volumes policy, for information on how to do this.

Reply:
Thank you for this comment!

I addressed this, now it's using validate each as other rightsize policies :)